tests/services/test_service_queryparser.py

```diff
@@ -96,21 +96,23 @@ def test_multimatch(parser, query):
 
 
 @pytest.mark.parametrize(
-    "query,transformed_query,query_type",
+    "query,transformed_query,query_type, allow_list",
     [
-        ("title:test", "metadata.title:test", "query_string"),
-        ("title:(test test)", "metadata.title:(test test)", "query_string"),
-        ("title:[1 TO 5]", "metadata.title:[1 TO 5]", "query_string"),
-        # Invalid field names results in multi_match
-        ("description:test", "description:test", "multi_match"),
-        ("metadata.title:test", "metadata.title:test", "multi_match"),
+        ("title:test", "metadata.title:test", "query_string", None),
+        ("title:(test test)", "metadata.title:(test test)", "query_string", None),
+        ("title:[1 TO 5]", "metadata.title:[1 TO 5]", "query_string", None),
+        # Term is allowed and not mapped, results in a query string with the original query
+        ("description:test", "description:test", "query_string", None),
+        ("metadata.title:test", "metadata.title:test", "query_string", None),
+        # Terms might be mapped but not allowed, results in multi_match with the original query
+        ("title:test", "title:test", "multi_match", ["metadata.date"]),
```

Comment:
I'm missing something (because tests pass). Why isn't this entering `transformer.py#44`? The query term `title` is in the mapping and there is an allowed list without it so:

```
if self._allow_list and not term_name in self._allow_list:

if ["metadata.date"] and not `title` in ["metadata.date"]:

True and not False:
```

I'm puzzled 😂

Reply:
https://github.com/inveniosoftware/invenio-records-resources/blob/0547e5d7cb7d30cca5c470e6fb264de408ce77ed/invenio_records_resources/services/records/queryparser/query.py#L92